src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import (division, print_function)
+
+import json
+import logging
+
+from WMCore.Services.Service import Service
+
+
+class DDMReqTemplate(dict):
+    """
+    A simple class defining a set of request templates for DDM, with default
+    structures representing the different POST calls expected by the modules
+    here:
+        https://github.com/SmartDataProjects/dynamo/tree/master/lib/web/modules
+    """
+
+    def __init__(self, api=None, *arg, **kw):
+        """
+        api:
+            String representing the type of request to be sent. In case the api
+            is missing or unrecognized the created object will be an ordinary
+            dictionary.
+        TODO:
+            To make the constructor fully dict like. E.g. I want to be able to
+            use it like:
+            ddmReq5 = DDMReqTemplate('copy',
+                                     item=['/LQLQToTop..._M-1200/NANOAODSIM'],
+                                     site=['T2_CH_CERN', 'T2_US_MIT'])
+        """
+        super(DDMReqTemplate, self).__init__(*arg, **kw)
+        if not api:
+            return
+        self.api = api
+
+        if api == 'copy':
+            self.update({'item': [],
+                         'site': ['T2_*', 'T1_*_Disk'],
+                         'group': 'DataOps',
+                         'n': None,
+                         'cache': None})
+
+        elif api == 'pollcopy':
+            self.update({'request_id': None,
+                         'item': None,
+                         'site': None,
+                         'status': None,
+                         'user': None})
+
+        elif api == 'cancelcopy':
+            self.update({'request_id': None})
+
+        else:
+            return
+
+    def strip(self):
+        """
+        _strip_
+
+        Method used to drop all the fields with value None. DDM does not
+        handle those quite well, and they better be removed from the request.
+        """
+
+        for key in list(self):
+            if not self[key]:
+                self.pop(key)
+
+    def isEqual(self, ddmReq, exclude=None):
```

Comment:
Hi @amaltaro , I was thinking of doing so, but in this case I need to exclude the aggregation key from the check since the objects will always differ on that one, so I need to pass it as a parameter to the check somehow. And with operators I do not know how  to do it. At the end decided to go for the simple but working solution.

Reply:
I think we will not have time to make it perfect - I suggest not to change this one. Resolving this conversation now. @amaltaro if you disagree, let me know.